virtcontainers/acrn.go

```diff
@@ -0,0 +1,621 @@
+}
+
+func (a *acrn) appendImage(devices []Device, imagePath string) ([]Device, error) {
+	if imagePath == "" {
+		return nil, fmt.Errorf("Image path is empty: %s", imagePath)
+	}
+
+	// Get sandbox and increment the globalIndex.
+	// This is to make sure the VM rootfs occupies
+	// the first Index which is /dev/vda.
+	sandbox, err := globalSandboxList.lookupSandbox(a.id)
+	if sandbox == nil && err != nil {
+		return nil, err
+	}
+	sandbox.GetAndSetSandboxBlockIndex()
+
+	devices, err = a.arch.appendImage(devices, imagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	return devices, nil
+}
+
+func (a *acrn) buildDevices(imagePath string) ([]Device, error) {
+	var devices []Device
+
+	if imagePath == "" {
+		return nil, fmt.Errorf("Image Path should not be empty: %s", imagePath)
+	}
+
+	console, err := a.getSandboxConsole(a.id)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add bridges before any other devices. This way we make sure that
+	// bridge gets the first available PCI address.
+	devices = a.arch.appendBridges(devices)
+
+	//Add LPC device to the list of other devices.
+	devices = a.arch.appendLPC(devices)
+
+	devices = a.arch.appendConsole(devices, console)
+
+	devices, err = a.appendImage(devices, imagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create virtio blk devices with dummy backend as a place
+	// holder for container rootfs (as acrn doesn't support hot-plug).
+	// Once the container rootfs is known, replace the dummy backend
+	// with actual path (using block rescan feature in acrn)
+	devices, err = a.createDummyVirtioBlkDev(devices)
+	if err != nil {
+		return nil, err
+	}
+
+	return devices, nil
+}
+
+// setup sets the Acrn structure up.
+func (a *acrn) setup(id string, hypervisorConfig *HypervisorConfig, vcStore *store.VCStore) error {
+	span, _ := a.trace("setup")
+	defer span.Finish()
+
+	err := hypervisorConfig.valid()
+	if err != nil {
+		return err
+	}
+
+	a.id = id
+	a.store = vcStore
+	a.config = *hypervisorConfig
+	a.arch = newAcrnArch(a.config)
+	if err = a.store.Load(store.Hypervisor, &a.state); err != nil {
+		// acrn currently supports only known UUIDs for security
+		// reasons (FuSa). When launching VM, only these pre-defined
+		// UUID should be used else VM launch will fail. acrn team is
+		// working on a solution to expose these pre-defined UUIDs
+		// to Kata in order for it to launch VMs successfully.
+		// Until this support is available, Kata is limited to
+		// launch 1 VM (using the default UUID).
+		// https://github.com/kata-containers/runtime/issues/1785
+
+		// The path might already exist, but in case of VM templating,
+		// we have to create it since the sandbox has not created it yet.
+		if err = os.MkdirAll(store.SandboxRuntimeRootPath(id), store.DirMode); err != nil {
+			return err
+		}
+
+		if err = a.store.Store(store.Hypervisor, a.state); err != nil {
+			return err
+		}
+	}
+
+	if err = a.store.Load(store.Hypervisor, &a.info); err != nil {
```

Comment:
I think you must save and load same information always, if you save/load `a.info` here and save/load `a.state` there, it will bring chaos.

If you plan to store both `a.state` and `a.info`, then you should merge them and save them together. 

@vijaydhanraj